lms/static/scripts/frontend_apps/components/EmailNotificationsPreferences.tsx

```diff
@@ -0,0 +1,96 @@
+import type { PanelProps } from '@hypothesis/frontend-shared';
+import { Button, Checkbox, Panel } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { StateUpdater } from 'preact/hooks';
+import { useCallback } from 'preact/hooks';
+
+import type { EmailNotificationsPreferences, WeekDay } from '../config';
+
+const dayNames: [WeekDay, string][] = [
+  [7, 'Sunday'],
+  [1, 'Monday'],
+  [2, 'Tuesday'],
+  [3, 'Wednesday'],
+  [4, 'Thursday'],
+  [5, 'Friday'],
+  [6, 'Saturday'],
+];
```

Comment:
Using an array of touples here instead of a Record, so that we can iterate over it and get proper type inference on both values.

Using `Object.values(dayNamesAsRecord)` ends up in the day being inferred as `string` and the name being inferred as `unknown`.